### PR TITLE
Update to v0.15.0

### PR DESCRIPTION
**Pull request by JordanMartinez**
**Description of the change**

Backlinking to purescript/purescript#4244

Migrates FFI to ES modules

---

**Checklist:**

- [ ] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0000)")
- [x] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,6 +12,8 @@ jobs:
       - uses: actions/checkout@v2
 
       - uses: purescript-contrib/setup-purescript@main
+        with:
+          purescript: "0.15.0-alpha-01"
```
Comment (thomashoneyman) [inline .github/workflows/ci.yml]: Should we just put this on unstable?

Comment (JordanMartinez) [reply .github/workflows/ci.yml]: D'oh!

**Comment by JordanMartinez** [comment]
This PR was created by running the following scripts from [purescript-ecosystem-update](https://github.com/JordanMartinez/purescript-ecosystem-update). Some of these scripts might not work on Mac computers:

[./init.sh](https://github.com/JordanMartinez/purescript-ecosystem-update/blob/master/init.sh)
[./forkAll.sh](https://github.com/JordanMartinez/purescript-ecosystem-update/blob/master/forkAll.sh) - this currently only forks the `prelude` and `newtype` library since I want to confirm that the changes I'm automating are correct
[./compile.sh 0 prelude](https://github.com/JordanMartinez/purescript-ecosystem-update/blob/master/compile.sh)
[./pr.sh 0 prelude](https://github.com/JordanMartinez/purescript-ecosystem-update/blob/master/pr.sh)

File: .eslintrc.json [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
   "env": {
-    "commonjs": true
+    "es6": true
   },
   "rules": {
-    "strict": [2, "global"],
+    "strict": [
+      2,
+      "global"
+    ],
```
Comment (JordanMartinez) [inline .eslintrc.json]: I'm using `jq` to do the updates in this file, but it causes this unneeded diff. Unless someone has a better idea for how to update the file, I think we just let these changes in.

File: src/Control/Apply.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
-"use strict";
 
-exports.arrayApply = function (fs) {
```
Comment (JordanMartinez) [inline src/Control/Apply.js:3]: My `sed` skills aren't good enough to know how to delete the `"use strict"` line followed by any empty lines. If someone else can provide a solution I can use in my script, let me know.

**Comment by JordanMartinez** [comment]
Oh, and [CI is failing](https://github.com/purescript/purescript-prelude/runs/5398662380?check_suite_focus=true#step:7:7) because `pulp` hasn't been migrated to work on ES modules yet.

**Comment by JordanMartinez** [comment]
We'll need to port https://github.com/purescript/spago/pull/846 and https://github.com/purescript/spago/pull/856 to `pulp`. However, `pulp` is still on Travis CI.

File: .eslintrc.json (hunk 2 above)
Comment (thomashoneyman) [reply .eslintrc.json]: I don't personally care about the changed diff much, but could we also just find/replace `commonjs` -> `es6` if that's the direction we're taking?

Comment (JordanMartinez) [reply .eslintrc.json]: True, that could work. However, we also have to add `sourceModule: "type"`, which does require knowing a bit more about the file. I could insert that content at a specific line. Let me try that out.

File: src/Control/Apply.js (hunk 3 above)
Comment (JordanMartinez) [reply src/Control/Apply.js:3]: Ok, I reverted to using a node script to handle the multi-line replacement.

File: .eslintrc.json (hunk 2 above)
Comment (JordanMartinez) [reply .eslintrc.json]: Inserting the line works, but it'll go wrong if any of the `.eslintrc.json` files are inconsistent.

**Comment by JordanMartinez** [comment]
I'm closing this PR. See https://github.com/working-group-purescript-es/purescript/issues/16

I've saved the work I've done here via the `es-modules-libraries` branch and `es-modules-continue` branch.

**Pull request closed without merging**